static/app/views/explore/contexts/spanTagsContext.tsx

```diff
@@ -89,27 +96,44 @@ export function SpanTagsProvider({children, dataset, enabled}: SpanTagsProviderP
     };
   }, [dataset, indexedTags, stringTags]);
 
-  const tags = useMemo(() => {
+  const tagsResult = useMemo(() => {
     return {
       number: allNumberTags,
       string: allStringTags,
+      numberTagsLoading,
+      stringTagsLoading,
     };
-  }, [allNumberTags, allStringTags]);
+  }, [allNumberTags, allStringTags, numberTagsLoading, stringTagsLoading]);
 
-  return <SpanTagsContext.Provider value={tags}>{children}</SpanTagsContext.Provider>;
+  return (
+    <SpanTagsContext.Provider value={tagsResult}>{children}</SpanTagsContext.Provider>
+  );
 }
 
 export function useSpanTags(type?: 'number' | 'string') {
-  const typedTags = useContext(SpanTagsContext);
+  const typedTagsResult = useContext(SpanTagsContext);
+
+  if (typedTagsResult === undefined) {
+    throw new Error('useSpanTags must be used within a SpanTagsProvider');
+  }
+
+  if (type === 'number') {
+    return typedTagsResult.number;
+  }
+  return typedTagsResult.string;
+}
+
+export function useSpanTagsAndStatus(type?: 'number' | 'string') {
```

Comment:
Might be worth just updating useSpanTags since this hook is effectively returning some additional data to `useSpanTags`